Index: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/upload/UIAddMetadataForm.java
===================================================================
--- core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/upload/UIAddMetadataForm.java	(revision 65274)
+++ core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/upload/UIAddMetadataForm.java	(working copy)
@@ -119,8 +119,12 @@
               valueList.add(uiJCRExplorer.getSession().getValueFactory().createValue(uiFormDateTime.getCalendar())) ;
               node.setProperty(name, valueList.toArray(new Value[] {}));
             } else {
-              List<String> values = (List<String>) ((UIFormMultiValueInputSet)uiForm.getUIInput(inputName)).getValue() ;
-              node.setProperty(name, values.toArray(new String[values.size()]));
+            	if (((UIFormInput)uiForm.getUIInput(inputName)) instanceof UIFormSelectBox){
+            		node.setProperty(name, ((UIFormSelectBox)uiForm.getUIInput(inputName)).getSelectedValues());
+            	}else {
+            		List<String> values=(List<String>) ((UIFormMultiValueInputSet)uiForm.getUIInput(inputName)).getValue() ;
+            		node.setProperty(name, values.toArray(new String[values.size()]));
+            	}              
             }
           } else {
             if (requiredType == 6) { // boolean
