diff --git a/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManagePublicationsActionComponent.java b/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManagePublicationsActionComponent.java
index 3f52932..e854cc8 100644
--- a/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManagePublicationsActionComponent.java
+++ b/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManagePublicationsActionComponent.java
@@ -29,6 +29,7 @@ import org.exoplatform.ecm.webui.component.explorer.control.filter.HasPublicatio
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsDocumentFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotLockedFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotRootNodeFilter;
+import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotIgnoreVersionNodeFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.listener.UIActionBarActionListener;
 import org.exoplatform.ecm.webui.component.explorer.popup.admin.UIActivePublication;
 import org.exoplatform.ecm.webui.component.explorer.popup.admin.UIPublicationManager;
@@ -64,7 +65,7 @@ public class ManagePublicationsActionComponent extends UIComponent {
       new IsDocumentFilter("UIActionBar.msg.manage-publication.not-supported-nodetype"),
       new IsNotRootNodeFilter("UIActionBar.msg.cannot-enable-publication-rootnode"),
       new CanSetPropertyFilter("UIActionBar.msg.access-denied"), new CanRemoveNodeFilter(),
-      new IsNotLockedFilter()                         });
+      new IsNotLockedFilter(), new IsNotIgnoreVersionNodeFilter()});
 
   @UIExtensionFilters
   public List<UIExtensionFilter> getFilters() {
diff --git a/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManageVersionsActionComponent.java b/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManageVersionsActionComponent.java
index 55881cb..dc51afb 100644
--- a/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManageVersionsActionComponent.java
+++ b/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManageVersionsActionComponent.java
@@ -28,6 +28,7 @@ import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotFolderFi
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotEditingDocumentFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotRootNodeFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotLockedFilter;
+import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotIgnoreVersionNodeFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.listener.UIActionBarActionListener;
 import org.exoplatform.ecm.webui.component.explorer.versions.UIActivateVersion;
 import org.exoplatform.ecm.webui.component.explorer.versions.UIVersionInfo;
@@ -56,7 +57,7 @@ public class ManageVersionsActionComponent extends UIComponent {
   private static final List<UIExtensionFilter> FILTERS = Arrays.asList(new UIExtensionFilter[] {
       new IsNotRootNodeFilter("UIActionBar.msg.cannot-enable-version-rootnode"),
       new CanSetPropertyFilter(), new IsNotLockedFilter(), new CanEnableVersionFilter(),
-      new IsNotEditingDocumentFilter(), new IsNotFolderFilter()
+      new IsNotEditingDocumentFilter(), new IsNotFolderFilter(), new IsNotIgnoreVersionNodeFilter()
       }
   );
 
diff --git a/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/upload/UIUploadForm.java b/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/upload/UIUploadForm.java
index 683b624..1ab9b70 100644
--- a/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/upload/UIUploadForm.java
+++ b/core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/upload/UIUploadForm.java
@@ -578,8 +578,13 @@ public class UIUploadForm extends UIForm implements UIPopupComponent, UISelectab
 
                 return ;
               }
-              if(!node.isNodeType(Utils.MIX_VERSIONABLE) && node.canAddMixin(Utils.MIX_VERSIONABLE)) {
-                node.addMixin(Utils.MIX_VERSIONABLE);
+              String nodetypes = System.getProperty("wcm.nodetypes.ignoreversion");
+              if(nodetypes == null || nodetypes.length() == 0)
+             	  nodetypes = "exo:webContent";
+              if(Utils.isMakeVersionable(node, nodetypes.split(","))) {
+              	if(!node.isNodeType(Utils.MIX_VERSIONABLE) && node.canAddMixin(Utils.MIX_VERSIONABLE)) {
+                  node.addMixin(Utils.MIX_VERSIONABLE);
+                }
               }
               Node contentNode = node.getNode(Utils.JCR_CONTENT);
               if(!node.isCheckedOut()) node.checkout() ;
diff --git a/core/webui/src/main/java/org/exoplatform/ecm/webui/utils/Utils.java b/core/webui/src/main/java/org/exoplatform/ecm/webui/utils/Utils.java
index 8e81bfc..4765cd5 100644
--- a/core/webui/src/main/java/org/exoplatform/ecm/webui/utils/Utils.java
+++ b/core/webui/src/main/java/org/exoplatform/ecm/webui/utils/Utils.java
@@ -1006,5 +1006,25 @@ public class Utils {
     }
     return null;
   }
+  
+  /* Check the current node is eligible to add mix:versionable or not
+   * 
+   * @param node the current node
+   * @param nodetypes The list of node types have child nodes which are not add mix:versionaboe while enrolling.
+   * @throws Exception the exception 
+   */
+  public static boolean isMakeVersionable(Node node, String[] nodetypes) throws Exception {
+  	boolean isMakeVersionable = true;  	
+  	int deep = node.getDepth();  	
+  	for(int i = 0; i< deep; i++) {
+  		Node parent = node.getParent();
+  		for (String nodetype: nodetypes) {
+  			if(nodetype != null && nodetype.length() > 0 && parent.isNodeType(nodetype))
+  				return false;
+  		}  		
+  	  node = parent;
+  	}    	
+  	return isMakeVersionable;
+  }
 
 }
diff --git a/delivery/wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties b/delivery/wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties
index 0db2bb6..9df35cf 100644
--- a/delivery/wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties
+++ b/delivery/wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties
@@ -157,3 +157,6 @@ wcm.cache.scriptservice.timetolive=600000
 # ECMS Cache Configuration - Webcontent
 wcm.cache.webcontent.initialwebcontentplugin.capacity=300
 wcm.cache.webcontent.initialwebcontentplugin.timetolive=600000
+
+# The node types which are ignored add "mix:versionable" while enrolling content 
+wcm.nodetypes.ignoreversion=exo:webContent
diff --git a/delivery/wkf-wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties b/delivery/wkf-wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties
index 23bbd0b..ccda6d0 100644
--- a/delivery/wkf-wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties
+++ b/delivery/wkf-wcm/server-tomcat/src/main/tomcat/gatein/conf/configuration.properties
@@ -155,4 +155,7 @@ wcm.cache.scriptservice.timetolive=600000
 
 # ECMS Cache Configuration - Webcontent
 wcm.cache.webcontent.initialwebcontentplugin.capacity=300
-wcm.cache.webcontent.initialwebcontentplugin.timetolive=600000
\ No newline at end of file
+wcm.cache.webcontent.initialwebcontentplugin.timetolive=600000
+
+# The node types which are ignored add "mix:versionable" while enrolling content 
+wcm.nodetypes.ignoreversion=exo:webContent
\ No newline at end of file
diff --git a/ext/authoring/services/src/main/java/org/exoplatform/services/wcm/extensions/publication/lifecycle/authoring/AuthoringPublicationPlugin.java b/ext/authoring/services/src/main/java/org/exoplatform/services/wcm/extensions/publication/lifecycle/authoring/AuthoringPublicationPlugin.java
index 5132501..b8bfa66 100644
--- a/ext/authoring/services/src/main/java/org/exoplatform/services/wcm/extensions/publication/lifecycle/authoring/AuthoringPublicationPlugin.java
+++ b/ext/authoring/services/src/main/java/org/exoplatform/services/wcm/extensions/publication/lifecycle/authoring/AuthoringPublicationPlugin.java
@@ -18,6 +18,7 @@ import org.exoplatform.container.PortalContainer;
 import org.exoplatform.portal.config.model.Page;
 import org.exoplatform.portal.mop.navigation.NavigationContext;
 import org.exoplatform.portal.webui.util.Util;
+import org.exoplatform.ecm.webui.utils.Utils;
 import org.exoplatform.services.cms.CmsService;
 import org.exoplatform.services.ecm.publication.IncorrectStateUpdateLifecycleException;
 import org.exoplatform.services.listener.ListenerService;
@@ -402,10 +403,15 @@ public class AuthoringPublicationPlugin extends StageAndVersionPublicationPlugin
    * org.exoplatform.services.ecm.publication.PublicationPlugin#addMixin(javax
    * .jcr.Node)
    */
-  public void addMixin(Node node) throws Exception {
-    node.addMixin(AuthoringPublicationConstant.PUBLICATION_LIFECYCLE_TYPE);
-    if (!node.isNodeType(AuthoringPublicationConstant.MIX_VERSIONABLE)) {
-      node.addMixin(AuthoringPublicationConstant.MIX_VERSIONABLE);
+  public void addMixin(Node node) throws Exception {  	
+    node.addMixin(AuthoringPublicationConstant.PUBLICATION_LIFECYCLE_TYPE);    
+    String nodetypes = System.getProperty("wcm.nodetypes.ignoreversion");
+    if(nodetypes == null || nodetypes.length() == 0)
+   	  nodetypes = "exo:webContent";
+    if(Utils.isMakeVersionable(node, nodetypes.split(","))) {
+	    if (!node.isNodeType(AuthoringPublicationConstant.MIX_VERSIONABLE)) {
+	      node.addMixin(AuthoringPublicationConstant.MIX_VERSIONABLE);
+	    }
     }
   }
 
