Index: apps/portlet-explorer/src/main/webapp/javascript/eXo/ecm/UIListView.js
===================================================================
--- apps/portlet-explorer/src/main/webapp/javascript/eXo/ecm/UIListView.js	(revision 77160)
+++ apps/portlet-explorer/src/main/webapp/javascript/eXo/ecm/UIListView.js	(working copy)
@@ -166,10 +166,14 @@
 					position: "absolute",
 					display: "none",
 					overflow: "hidden",
+					padding: "1px",
 					background: "white",
-					width: element.offsetWidth + "px"
+					border: "1px solid gray",
+					width: element.offsetWidth + 50 + "px",
+					height: "25px"
 				}
 			});
+			eXo.core.Browser.setOpacity(mobileElement, 65);
 			Self.mobileId = mobileElement.id;
 			var coverElement = newElement({
 				className: "UITreeExplorer",
Index: apps/portlet-explorer/src/main/webapp/javascript/eXo/ecm/UISimpleView.js
===================================================================
--- apps/portlet-explorer/src/main/webapp/javascript/eXo/ecm/UISimpleView.js	(revision 77160)
+++ apps/portlet-explorer/src/main/webapp/javascript/eXo/ecm/UISimpleView.js	(working copy)
@@ -139,10 +139,14 @@
 					position: "absolute",
 					display: "none",
 					overflow: "hidden",
+					padding: "1px",
 					background: "white",
-					width: element.offsetWidth + "px"
+					border: "1px solid gray",
+					width: element.offsetWidth + 50 + "px",
+					height: "25px"
 				}
 			});
+			eXo.core.Browser.setOpacity(mobileElement, 65);
 			Self.mobileId = mobileElement.id;
 			var coverElement = newElement({
 				className: "UITreeExplorer",
