Index: core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java
===================================================================
--- core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java	(revision 66125)
+++ core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java	(working copy)
@@ -28,6 +28,7 @@
 import javax.jcr.Property;
 import javax.jcr.PropertyType;
 
+import org.exoplatform.commons.utils.IOUtil;
 import org.exoplatform.ecm.webui.form.validator.CronExpressionValidator;
 import org.exoplatform.ecm.webui.form.validator.ECMNameValidator;
 import org.exoplatform.ecm.webui.form.validator.RepeatCountValidator;
@@ -108,8 +109,9 @@
           if (input instanceof UIFormUploadInput) {
         	  UploadResource uploadResource = ((UIFormUploadInput) input).getUploadResource();
         	  if (uploadResource == null) continue;
-        	    inputStream = ((UIFormUploadInput) input).getUploadDataAsStream();
-              property.setValue(inputStream);
+        	  String location = uploadResource.getStoreLocation();
+      	      byte[] uploadData = IOUtil.getFileContentAsBytes(location);
+      	      property.setValue(uploadData);
 
               mimeTypeJcrPath = property.getJcrPath().replace("jcr:data", "jcr:mimeType");
               JcrInputProperty mimeTypeInputPropertyTmp = new JcrInputProperty();
