Index: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java
===================================================================
--- core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java	(revision 70911)
+++ core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java	(working copy)
@@ -115,12 +115,12 @@
   
   private PortalContainer manager;
   
-  private OrganizationService organizationService; 
+  private OrganizationService organizationService=null; 
   
-  private ResourceBundleService resourceBundleService;
+  private ResourceBundleService resourceBundleService=null;
   
   private String resourceBundleNames[];
-  private ResourceBundle sharedResourceBundle;
+  private ResourceBundle sharedResourceBundle=null;
   
   private Locale lang = Locale.ENGLISH;
   /**
@@ -132,10 +132,6 @@
   public DriverConnector(InitParams params) {
     limit = Integer.parseInt(params.getValueParam("upload.limit.size").getValue());
     manager = PortalContainer.getInstance() ;
-    organizationService = WCMCoreUtils.getService(OrganizationService.class);
-    resourceBundleService = WCMCoreUtils.getService(ResourceBundleService.class);
-    resourceBundleNames = resourceBundleService.getSharedResourceBundleNames();
-    sharedResourceBundle = resourceBundleService.getResourceBundle(resourceBundleNames, lang);
   }
 	
   /**
@@ -403,6 +399,11 @@
   }
   
   private String resolveDriveLabel(String name, String lang) {
+  	if (resourceBundleService ==null) {
+  		resourceBundleService = WCMCoreUtils.getService(ResourceBundleService.class);
+  		resourceBundleNames = resourceBundleService.getSharedResourceBundleNames();
+  		sharedResourceBundle = resourceBundleService.getResourceBundle(resourceBundleNames, this.lang);
+  	}
 	  try {
 		  if(!this.lang.getLanguage().equals(lang)){
 			  this.lang = new Locale(lang);
@@ -500,6 +501,9 @@
    * @throws Exception the exception
    */
   private List<String> getMemberships(String userId) throws Exception {
+  	if (organizationService==null) {
+  		organizationService = WCMCoreUtils.getService(OrganizationService.class);
+  	}
   	((ComponentRequestLifecycle) organizationService).startRequest(manager);
     List<String> userMemberships = new ArrayList<String> ();
     userMemberships.add(userId);
@@ -525,6 +529,9 @@
    * @throws Exception the exception
    */
   private List<String> getGroups(String userId) throws Exception {
+  	if (organizationService==null) {
+  		organizationService = WCMCoreUtils.getService(OrganizationService.class);
+  	}
     ((ComponentRequestLifecycle) organizationService).startRequest(manager);
     List<String> groupList = new ArrayList<String> ();
     Collection<?> groups = organizationService.getGroupHandler().findGroupsOfUser(userId);
