Index: ext/authoring/services/src/main/resources/groovy/wcm/webui/publication/lifecycle/authoring/ui/UIPublicationPanel.gtmpl
===================================================================
--- ext/authoring/services/src/main/resources/groovy/wcm/webui/publication/lifecycle/authoring/ui/UIPublicationPanel.gtmpl	(revision 71892)
+++ ext/authoring/services/src/main/resources/groovy/wcm/webui/publication/lifecycle/authoring/ui/UIPublicationPanel.gtmpl	(working copy)
@@ -1,6 +1,7 @@
 <%
 	import java.util.List;
-    import java.util.ArrayList;
+  import java.util.ArrayList;
+  import org.exoplatform.ecm.utils.text.Text;
 	import javax.jcr.Node;
 	import javax.jcr.version.Version;
 	import org.exoplatform.services.wcm.extensions.publication.lifecycle.authoring.ui.UIPublicationPanel;
@@ -38,8 +39,10 @@
   <% } %>
 
   <fieldset class="StatusTable">
-    <% String versionName = currentRevision.getName();  %>
-    <legend class="Legend"> <%=_ctx.appRes("UIPublicationPanel.status.version")%> $versionName </legend>
+     <% String versionName = currentRevision.getName();
+        String verName = Text.unescapeIllegalJcrChars(versionName);
+     %>
+     <legend class="Legend"> <%=_ctx.appRes("UIPublicationPanel.status.version")%> $verName </legend>
     <div class="StatusAction">
       <div class="StatusTitle"><%=_ctx.appRes("UIPublicationPanel.status.title")%>: </div>
       <div class="StatusLink">
@@ -126,11 +129,11 @@
               <%
                 if (versionName.equals(revision.getName())) {
                   %>
-                    <a><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= revision.getName() %></a>
+                    <a><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= Text.unescapeIllegalJcrChars(revision.getName()) %></a>
                   <%
                 } else {
                   %>
-                    <a class="ActiveLink" href="<%= uicomponent.event("ChangeVersion", revision.getUUID()) %>"><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= revision.getName() %></a>
+                    <a class="ActiveLink" href="<%= uicomponent.event("ChangeVersion", revision.getUUID()) %>"><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= Text.unescapeIllegalJcrChars(revision.getName()) %></a>
                   <%
                 }
               %>
Index: core/publication-plugins/src/main/resources/groovy/wcm/webui/publication/lifecycle/stageversion/ui/UIPublicationPanel.gtmpl
===================================================================
--- core/publication-plugins/src/main/resources/groovy/wcm/webui/publication/lifecycle/stageversion/ui/UIPublicationPanel.gtmpl	(revision 71892)
+++ core/publication-plugins/src/main/resources/groovy/wcm/webui/publication/lifecycle/stageversion/ui/UIPublicationPanel.gtmpl	(working copy)
@@ -1,5 +1,6 @@
 <%
   import java.util.List;
+  import org.exoplatform.ecm.utils.text.Text;
   import javax.jcr.Node;
   import javax.jcr.version.Version;
   import org.exoplatform.services.wcm.publication.lifecycle.stageversion.ui.UIPublicationPanel;
@@ -14,8 +15,10 @@
 <% uicomponent.begin(); %>
 <div class="UIPublicationPanel" id="$uicomponent.id">
   <fieldset class="StatusTable">
-    <% String versionName = currentRevision.getName();  %>
-    <legend class="Legend"> <%=_ctx.appRes("UIPublicationPanel.status.version")%> $versionName </legend>
+     <% String versionName = (currentRevision.getName();
+        String verName = Text.unescapeIllegalJcrChars(versionName);
+     %>
+     <legend class="Legend"> <%=_ctx.appRes("UIPublicationPanel.status.version")%> $verName </legend>
     <div class="StatusAction">
       <div class="StatusTitle"><%=_ctx.appRes("UIPublicationPanel.status.title")%>: </div>
       <div class="StatusLink">
@@ -95,11 +98,11 @@
               <%
                 if (versionName.equals(revision.getName())) {
                   %>
-                    <a><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= revision.getName() %></a>
+                    <a><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= Text.unescapeIllegalJcrChars(revision.getName()) %></a>
                   <%
                 } else {
                   %>
-                    <a class="ActiveLink" href="<%= uicomponent.event("ChangeVersion", revision.getUUID()) %>"><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= revision.getName() %></a>
+                    <a class="ActiveLink" href="<%= uicomponent.event("ChangeVersion", revision.getUUID()) %>"><%=_ctx.appRes("UIPublicationPanel.revisions.label")%>: <%= Text.unescapeIllegalJcrChars(revision.getName()) %></a>
                   <%
                 }
               %>
Index: core/publication-plugins/src/main/resources/groovy/wcm/webui/publication/lifecycle/stageversion/ui/UIPublicationHistory.gtmpl
===================================================================
--- core/publication-plugins/src/main/resources/groovy/wcm/webui/publication/lifecycle/stageversion/ui/UIPublicationHistory.gtmpl	(revision 71892)
+++ core/publication-plugins/src/main/resources/groovy/wcm/webui/publication/lifecycle/stageversion/ui/UIPublicationHistory.gtmpl	(working copy)
@@ -7,6 +7,7 @@
  * 9:36:26 AM 
  */
   import java.text.SimpleDateFormat;
+  import org.exoplatform.ecm.utils.text.Text;
   import org.exoplatform.commons.utils.ISO8601;
   import org.exoplatform.services.wcm.publication.lifecycle.stageversion.ui.UIPublicationContainer;
 %>
@@ -33,7 +34,7 @@
   	      out.println("<tr class=\"" + rowClass + "\">");
   	      
 					out.println("<td><div class=\"Text\">" + uicomponent.getAncestorOfType(UIPublicationContainer.class).getDateTimeFormater().format(bean.getLogDate().getTime()) + "</div></td>");
-					out.println("<td><div class=\"Text\">" + bean.getVersionName() + "</div></td>");
+					out.println("<td><div class=\"Text\">" + Text.unescapeIllegalJcrChars(bean.getVersionName()) + "</div></td>");
 					out.println("<td><div class=\"Text\">" + _ctx.appRes("UIPublicationPanel.status." + bean.getState()) + "</div></td>");	          
 					out.println("<td><div class=\"Text\">" + bean.getAuthor() + "</div></td>");	          
 					out.println("<td><div class=\"Text\">" + _ctx.appRes(bean.getDescription()) + "</div></td>");	          
