Index: core/services/src/main/java/org/exoplatform/services/cms/webdav/WebDavServiceImpl.java
===================================================================
--- core/services/src/main/java/org/exoplatform/services/cms/webdav/WebDavServiceImpl.java	(revision 79245)
+++ core/services/src/main/java/org/exoplatform/services/cms/webdav/WebDavServiceImpl.java	(working copy)
@@ -357,6 +357,7 @@
                       @HeaderParam(ExtHttpHeaders.CONTENT_NODETYPE) String nodeTypeHeader,
                       @HeaderParam(ExtHttpHeaders.CONTENT_MIXINTYPES) String mixinTypes,
                       @HeaderParam(ExtHttpHeaders.CONTENTTYPE) MediaType mediaType,
+                      @HeaderParam(ExtHttpHeaders.USER_AGENT) String userAgent,
                       InputStream inputStream) {
 
     Session session = null;
@@ -372,7 +373,7 @@
       log.warn("Cannot find the item at " + repoName + "/" + repoPath, e);
       return Response.serverError().build();
     }
-    Response res =  super.put(repoName, repoPath, lockTokenHeader, ifHeader, null, nodeTypeHeader, mixinTypes, mediaType, inputStream);
+    Response res =  super.put(repoName, repoPath, lockTokenHeader, ifHeader, null, nodeTypeHeader, mixinTypes, mediaType, userAgent, inputStream);
     try {
       Node currentNode = (Node) session.getItem(path(repoPath));
       listenerService.broadcast(this.POST_UPLOAD_CONTENT_EVENT, this, currentNode);      
