Index: apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/popup/info/UIPropertyTab.gtmpl
===================================================================
--- apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/popup/info/UIPropertyTab.gtmpl	(revision 71972)
+++ apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/popup/info/UIPropertyTab.gtmpl	(working copy)
@@ -6,6 +6,7 @@
 	 * September 14, 2006
 	 * 2:10:23 PM
 	 */
+import org.apache.commons.lang.StringEscapeUtils;
 %>
 <div class="UIPropertyTab">
   <table class="UIGrid" style="width: 97%">
@@ -56,8 +57,8 @@
 				      <%=_ctx.appRes("UIPropertyTab.label." + value)%>
 				    </div>
 				    <%} else {%>
-				    <div class="Text" title="<%=value%>">
-				      <%=value%>
+				    <div class="Text" title="<%=StringEscapeUtils.escapeHtml(value)%>">
+				      <%=StringEscapeUtils.escapeHtml(value)%>
 				   	</div>
 				    <%}%>
 			    </td>
