Index: core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java
===================================================================
--- core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java	(revision 70331)
+++ core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java	(working copy)
@@ -48,6 +48,7 @@
 import org.exoplatform.webui.form.validator.NullFieldValidator;
 import org.exoplatform.webui.form.validator.NumberFormatValidator;
 import org.exoplatform.webui.form.validator.StringLengthValidator;
+import org.exoplatform.wcm.webui.Utils;
 
 /*
  * Created by The eXo Platform SAS
@@ -130,7 +131,12 @@
           } else if(input instanceof UIFormCheckBoxInput) {
             property.setValue(((UIFormCheckBoxInput)input).isChecked()) ;
           } else {
-            property.setValue(input.getValue()) ;
+          	if(input.getValue()!=null) {
+          		String inputValue = input.getValue().toString();          		
+          		property.setValue(Utils.sanitize(inputValue));
+          	} else {
+          		property.setValue(input.getValue());
+          	}
           }
         }
       }
Index: core/webui/src/main/java/org/exoplatform/wcm/webui/Utils.java
===================================================================
--- core/webui/src/main/java/org/exoplatform/wcm/webui/Utils.java	(revision 70331)
+++ core/webui/src/main/java/org/exoplatform/wcm/webui/Utils.java	(working copy)
@@ -643,6 +643,13 @@
   {
      return Util.getUIPortal().getSelectedNavigation();
   }
+  
+  public static String sanitize(String string) {
+    return string
+       .replaceAll("(?i)<script.*?>.*?</script.*?>", "")   // case 1 : <script> are removed 
+       .replaceAll("(?i)<.*?javascript:.*?>.*?</.*?>", "") // case 2 : javascript: call are removed
+       .replaceAll("(?i)<.*?\\s+on.*?>.*?</.*?>", "");     // case 3: remove on* attributes like onLoad or onClick
+  }
 
   
 }
