Index: core/publication/src/main/java/org/exoplatform/services/wcm/publication/MessageDigester.java
===================================================================
--- core/publication/src/main/java/org/exoplatform/services/wcm/publication/MessageDigester.java	(revision 73958)
+++ core/publication/src/main/java/org/exoplatform/services/wcm/publication/MessageDigester.java	(working copy)
@@ -28,7 +28,6 @@
         java.security.MessageDigest msgDigest = java.security.MessageDigest.getInstance("MD5");
         msgDigest.update(message.getBytes());
         byte[] aMessageDigest = msgDigest.digest();
-        // C'est pas beau mais on ne l'a pas décidé ;-)
         StringBuffer ticket = new StringBuffer();
         ticket = new StringBuffer();
         String tmp = null;
