Index: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIJCRExplorerPortlet.java
===================================================================
--- core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIJCRExplorerPortlet.java	(revision 71277)
+++ core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIJCRExplorerPortlet.java	(working copy)
@@ -197,9 +197,13 @@
     RenderResponse response = context.getResponse();
     Element elementS = response.createElement("script");
     elementS.setAttribute("type", "text/javascript");
-    elementS.setAttribute("src", "/eXoWCMResources/javascript/eXo/wcm/backoffice/public/Components.js");
+    elementS.setAttribute("src", "/eXoWCMResources/javascript/eXo/wcm/backoffice/public/Components.js");        
+    response.addProperty(MimeResponse.MARKUP_HEAD_ELEMENT,elementS);
+    
+    elementS = response.createElement("script");
+    elementS.setAttribute("type", "text/javascript");
     elementS.setAttribute("src", "/eXoWCMResources/javascript/eXo/wcm/frontoffice/public/jquery-1.5.1.js");
-    response.addProperty(MimeResponse.MARKUP_HEAD_ELEMENT,elementS); 
+    response.addProperty(MimeResponse.MARKUP_HEAD_ELEMENT,elementS);
     
     super.processRender(app, context);
   }
