Index: core/webui/src/main/java/org/exoplatform/wcm/webui/Utils.java
===================================================================
--- core/webui/src/main/java/org/exoplatform/wcm/webui/Utils.java	(revision 73535)
+++ core/webui/src/main/java/org/exoplatform/wcm/webui/Utils.java	(working copy)
@@ -693,8 +693,8 @@
     return string
        .replaceAll("(?i)<script.*?>.*?</script.*?>", "")   // case 1 : <script> are removed 
        .replaceAll("(?i)<.*?javascript:.*?>.*?</.*?>", "") // case 2 : javascript: call are removed
-       .replaceAll("eval\\((.*)\\)", "")                   // case 3 : eval: are removed
-       .replaceAll("'", "#39;");                         // case 4 : replace "'" by #39
+       .replaceAll("eval\\((.*)\\)", "");                   // case 3 : eval: are removed           
+//       .replaceAll("'", "#39;");                         // case 4 : replace "'" by #39
   }
 
   
