Index: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/popup/actions/UIDocumentForm.java
===================================================================
--- core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/popup/actions/UIDocumentForm.java	(revision 68615)
+++ core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/popup/actions/UIDocumentForm.java	(working copy)
@@ -574,16 +574,14 @@
             if ((clickedField != null) && (clickedField.equals(FIELD_TAXONOMY))){
               UIJCRExplorer uiExplorer = uiDocumentForm.getAncestorOfType(UIJCRExplorer.class);
               String repository = uiExplorer.getRepositoryName();
-              DMSConfiguration dmsConfig = uiDocumentForm.getApplicationComponent(DMSConfiguration.class);
-              DMSRepositoryConfiguration dmsRepoConfig = dmsConfig.getConfig(repository);
-              String workspaceName = dmsRepoConfig.getSystemWorkspace();            
               if(uiSet.getValue().size() == 0) uiSet.setValue(new ArrayList<Value>());            
               UIOneTaxonomySelector uiOneTaxonomySelector = 
                 uiFormController.createUIComponent(UIOneTaxonomySelector.class, null, null);
-              uiOneTaxonomySelector.setIsDisable(workspaceName, false);
               TaxonomyService taxonomyService = uiDocumentForm.getApplicationComponent(TaxonomyService.class);
               List<Node> lstTaxonomyTree = taxonomyService.getAllTaxonomyTrees(repository);
               if (lstTaxonomyTree.size() == 0) throw new AccessDeniedException();
+              String workspaceName = lstTaxonomyTree.get(0).getSession().getWorkspace().getName();
+              uiOneTaxonomySelector.setIsDisable(workspaceName, false);
               uiOneTaxonomySelector.setRootNodeLocation(repository, workspaceName, lstTaxonomyTree.get(0).getPath());
               uiOneTaxonomySelector.setExceptedNodeTypesInPathPanel(new String[] {Utils.EXO_SYMLINK});
               uiOneTaxonomySelector.init(uiExplorer.getSystemProvider());
