Index: ext/authoring/webui/src/main/java/org/exoplatform/wcm/extensions/ui/PublicationPublishAction.java
===================================================================
--- ext/authoring/webui/src/main/java/org/exoplatform/wcm/extensions/ui/PublicationPublishAction.java	(revision 69890)
+++ ext/authoring/webui/src/main/java/org/exoplatform/wcm/extensions/ui/PublicationPublishAction.java	(working copy)
@@ -28,6 +28,7 @@
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsCheckedOutFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotLockedFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.listener.UIActionBarActionListener;
+import org.exoplatform.ecm.webui.utils.LockUtil;
 import org.exoplatform.services.ecm.publication.PublicationService;
 import org.exoplatform.webui.config.annotation.ComponentConfig;
 import org.exoplatform.webui.config.annotation.EventConfig;
@@ -65,9 +66,14 @@
 			UIJCRExplorer uiExplorer = event.getSource().getAncestorOfType(UIJCRExplorer.class);
 			PublicationService publicationService = (PublicationService)PortalContainer.getInstance().getComponentInstanceOfType(PublicationService.class);
 			Node node = uiExplorer.getCurrentNode();
-		    HashMap<String,String> context = new HashMap<String,String>();
 			
-		    publicationService.changeState(node, "published", context);
+			if(node.isLocked()) {
+        node.getSession().addLockToken(LockUtil.getLockToken(node));
+      }
+		    
+			HashMap<String,String> context = new HashMap<String,String>();
+			
+		  publicationService.changeState(node, "published", context);
 
 			UIPopupContainer UIPopupContainer = uiExplorer.getChild(UIPopupContainer.class);
 			event.getRequestContext().addUIComponentToUpdateByAjax(UIPopupContainer);
