Index: apps/resources-wcm/src/main/webapp/javascript/eXo/wcm/backoffice/private/UIFormGeneratorPortlet.js
===================================================================
--- apps/resources-wcm/src/main/webapp/javascript/eXo/wcm/backoffice/private/UIFormGeneratorPortlet.js	(revision 77328)
+++ apps/resources-wcm/src/main/webapp/javascript/eXo/wcm/backoffice/private/UIFormGeneratorPortlet.js	(working copy)
@@ -56,8 +56,12 @@
 			fieldComponent  +=		"<td class='FieldComponent'><div class='CheckboxButton'><input type='checkbox' class='CheckBox' value='checkbox1'/><span style='padding : 0 5px 0 19px; display:block; line-height:12px'>Checkbox 1</span><div style='clear:left'></div></div></td>";
 			break;						
 		case "radio"		: 
+		  var groupId = "radiogroup" + UIFormGeneratorPortlet.prototype.getRandomInt(1,100);		  
+		  while(document.getElementsByName(groupId) > 0) {
+		    groupId = "radiogroup" + UIFormGeneratorPortlet.prototype.getRandomInt(1,100);
+		  }
 			fieldComponent  +=		"<td class='FieldLabel' value='Radio'>Radio field</td>";
-			fieldComponent  +=		"<td class='FieldComponent'><div class='RadioButton' idx='1'><input type='radio' class='Radio' value='radio1'/><span style='padding : 0 5px 0 19px; display:block; line-height:12px'>Radio 1</span><div style='clear:left'></div></div></td>";
+			fieldComponent  +=		"<td class='FieldComponent'><div class='RadioButton' idx='1'><input type='radio' name='"+groupId+"' class='Radio' value='radio1'/><span style='padding : 0 5px 0 19px; display:block; line-height:12px'>Radio 1</span><div style='clear:left'></div></div></td>";
 			multivalue		= true;
 			break;			
 		case "datetime"	: 
@@ -448,10 +452,12 @@
 				}
 			}						
 			break;
-		case "radio" :				
+		case "radio" :			  		
+		  var radioElement = DOMUtil.findFirstDescendantByClass(ancestorNode, 'input', 'Radio');
+		  var radioName = radioElement.getAttribute('name');
 			var radioNode  = document.createElement("div");
 			radioNode.setAttribute("idx",index);
-			radioNode.innerHTML = '<input type="radio" class="Radio" value="radio'+index+'" /><span style="padding : 0 5px 0 19px; display:block; line-height:12px">Radio '+index+'</span><div style="clear:left"></div>';
+			radioNode.innerHTML = '<input type="radio" name="'+radioName+'" class="Radio" value="radio'+index+'" /><span style="padding : 0 5px 0 19px; display:block; line-height:12px">Radio '+index+'</span><div style="clear:left"></div>';
 			radioNode.className = "RadioButton";					
 			for(var j =0; j< radioContainer.length; j++){			
 				if(radioContainer[j].getAttribute("idx")== currentIndex){
@@ -663,6 +669,10 @@
 	return strObject;
 };
 
+UIFormGeneratorPortlet.prototype.getRandomInt = function(min, max) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
 UIFormGeneratorPortlet.prototype.submitForm = function() {
 	var strJsonObject = eXo.ecm.UIFormGeneratorPortlet.getStringJsonObject();
 	var inputHidden = document.getElementById("UIFormGeneratorJsonObjectStringInput");
