Index: core/services/src/main/java/org/exoplatform/services/wcm/images/RESTImagesRendererService.java
===================================================================
--- core/services/src/main/java/org/exoplatform/services/wcm/images/RESTImagesRendererService.java	(revision 71567)
+++ core/services/src/main/java/org/exoplatform/services/wcm/images/RESTImagesRendererService.java	(working copy)
@@ -149,11 +149,11 @@
   + */
   private Date getLastModifiedDate(Node node) throws Exception {
      Date lastModifiedDate = null;
-     if (node.hasNode("jcr:content")) {
+     if (node.hasNode("jcr:content") && node.getNode("jcr:content").hasProperty("jcr:lastModified")) {
        lastModifiedDate = node.getNode("jcr:content").getProperty("jcr:lastModified").getDate().getTime();
      } else if (node.hasProperty("exo:dateModified")) {
          lastModifiedDate = node.getProperty("exo:dateModified").getDate().getTime();
-     } else {
+     } else if (node.hasProperty("jcr:created")){
        lastModifiedDate = node.getProperty("jcr:created").getDate().getTime();
      }
      return lastModifiedDate;
@@ -175,7 +175,8 @@
      Date lastModifiedDate = getLastModifiedDate(node);
     
      // Check if cached resource has not been modifed, return 304 code
-     if (ifModifiedSinceDate.getTime() >= lastModifiedDate.getTime()) {      
+     if (lastModifiedDate != null && ifModifiedSinceDate != null && 
+         ifModifiedSinceDate.getTime() >= lastModifiedDate.getTime()) {      
        return false;
      }   
      return true;
