Index: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationBigImage.gtmpl
===================================================================
--- packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationBigImage.gtmpl	(revision 73897)
+++ packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationBigImage.gtmpl	(working copy)
@@ -82,7 +82,7 @@
 							<div class="CaptionContent">				
 								<% 
 									if (itemSummary != null && itemSummary.trim().length() != 0 && uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_SUMMARY)) {
-										%><pre><%=itemSummary%></pre><% 
+										%><div><%=itemSummary%></div><% 
 									} 
 								%>															
 							</div>			
Index: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationDefault.gtmpl
===================================================================
--- packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationDefault.gtmpl	(revision 73897)
+++ packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationDefault.gtmpl	(working copy)
@@ -77,11 +77,13 @@
 							%>											
 							<div class="CaptionContent">				
 								<% 
+								  String marginstyle = "";
 									if(imgSrc != null && uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_ILLUSTRATION)) {
-										%><img src="$imgSrc" width="91" /><% 
+										marginstyle = "style=\"margin-left:95px;\"";
+										%><img src="$imgSrc" width="91" style="float:left" /><% 
 									}
 									if (itemSummary != null && itemSummary.trim().length() != 0 && uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_SUMMARY)) { 
-										%><pre><%=itemSummary%></pre><% 
+										%><div <%=marginstyle%> ><%=itemSummary%></div><div style="clear:both"/><% 
 									} 
 								%>															
 							</div>
@@ -92,6 +94,7 @@
               <%
               }
               %>
+              
 						</div>
 					<%							
 				} //end for
Index: packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationSmall.gtmpl
===================================================================
--- packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationSmall.gtmpl	(revision 73897)
+++ packaging/wcm/webapp/src/main/webapp/WEB-INF/conf/wcm-artifacts/application-templates/content-list-viewer/list/UIContentListPresentationSmall.gtmpl	(working copy)
@@ -58,7 +58,9 @@
 					%>
 					<%=uicomponent.addQuickEditDiv("CaptionItem", viewNode)%>
 				<%
+								String marginstyle = "";
                 if(imgSrc != null && uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_ILLUSTRATION)) {
+                	marginstyle = "style=\"margin-left:34px;\"";
                     %><img src="$imgSrc" width="30" style="float:left; padding-right:5px;" /><%
                 }
 								if (uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_TITLE)) {
@@ -81,9 +83,10 @@
 							<div class="CaptionContent">				
 								<% 
 									if (itemSummary != null && itemSummary.trim().length() != 0 && uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_SUMMARY)) {
-										%><pre><%=itemSummary%></pre><% 
+										%><div <%=marginstyle%> ><%=itemSummary%></div><% 
 									} 
-								%>															
+								%>
+								<div style="clear:both"></div>
 							</div>			
 	            <%
 	            if (uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_READMORE)) {
