Index: core/publication/src/main/java/org/exoplatform/services/wcm/publication/WCMComposerImpl.java
===================================================================
--- core/publication/src/main/java/org/exoplatform/services/wcm/publication/WCMComposerImpl.java	(revision 69576)
+++ core/publication/src/main/java/org/exoplatform/services/wcm/publication/WCMComposerImpl.java	(working copy)
@@ -204,7 +204,7 @@
 		NodeIterator nodeIterator = getViewableContents(repository, workspace, path, filters, sessionProvider);
 		List<Node> nodes = new ArrayList<Node>();
 		Node node = null, viewNode = null;
-		while (nodeIterator.hasNext()) {
+		while (nodeIterator != null && nodeIterator.hasNext()) {
 			node = nodeIterator.nextNode();
 			viewNode = getViewableContent(node, filters);
 			if (viewNode != null) {
@@ -240,11 +240,18 @@
 			addUsedPrimaryTypes(primaryType);
 			if (primaryType == null) {
 				primaryType = "nt:base";
-				Node currentFolder = "/".equals(path) ? session.getRootNode() : 
-				                                        session.getRootNode().getNode(path.substring(1));
-				if (currentFolder.isNodeType("exo:taxonomy")) {
-					primaryType = "exo:taxonomyLink";
-				}
+		        Node currentFolder = null;
+		        if ("/".equals(path)) {
+		          currentFolder = session.getRootNode();
+		        } else if (session.getRootNode().hasNode(path.substring(1))) {
+		          currentFolder = session.getRootNode().getNode(path.substring(1));
+		        } else {
+		          return null;
+		        }
+		               
+		        if (currentFolder != null && currentFolder.isNodeType("exo:taxonomy")) {
+		          primaryType = "exo:taxonomyLink";
+		        }
 			} else {
 				filterTemplates = false;
 			}
